### PR TITLE
Open and Close Functions

### PR DESCRIPTION
**Pull request by benJrohrer**

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
The style in this PR agrees with [`black`](https://github.com/ambv/black). :heavy_check_mark:

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: cirrus/google_cloud/manager.py [hunk 1]
```diff
@@ -104,6 +104,7 @@ def __init__(self, project_id=None, creds=None, use_default=True):
         )
         creds = creds or config.GOOGLE_APPLICATION_CREDENTIALS
         self.credentials = ServiceAccountCredentials.from_service_account_file(creds)
+        self.open_count = 0
```
Comment (Avantol13) [inline cirrus/google_cloud/manager.py]: lets name this _open_count with a leading underscore so people don't try to use this. or __enter_count__, that seems excessive though

File: cirrus/google_cloud/manager.py [hunk 2]
```diff
@@ -153,6 +154,40 @@ def __exit__(self, exception_type, exception_value, traceback):
         self._admin_service = None
         self._storage_client = None
 
+    def open(self):
+        """
+        Run initialization code in __enter__, but do not return self
+        Used for initializing GCM without using `with` block
+
+        Args:
```
Comment (Avantol13) [inline cirrus/google_cloud/manager.py]: You can remove these if there aren't args or returns

File: cirrus/google_cloud/manager.py [hunk 3]
```diff
@@ -153,6 +154,40 @@ def __exit__(self, exception_type, exception_value, traceback):
...
+            -
+        Return:
+            -
+
+        """
+        if self.open_count == 0:
+            self.__enter__()
+        self.open_count += 1
+        return
```
Comment (Avantol13) [inline cirrus/google_cloud/manager.py]: Don't need an explicit return if we're not returning anything, Python will return None by default

File: cirrus/google_cloud/manager.py (hunk 1 above)
Comment (Avantol13) [reply cirrus/google_cloud/manager.py]: I would add a quick clarification comment here too. `This allows for safely re-opening the object multiple times and prevents issues with attempting to close more than needed`. something like that.

File: cirrus/google_cloud/manager.py [hunk 4]
```diff
@@ -153,6 +156,29 @@ def __exit__(self, exception_type, exception_value, traceback):
...
+        Used for closing GCM without using `with` block
+        """
+        if self._open_count > 0:
+            self._open_count -= 1
+            if self._open_count == 0:
+                self._authed_session.close()
+                self._authed_session = None
+                self._admin_service = None
+                self._storage_client = None
+                self.is_open = False
```
Comment (Avantol13) [inline cirrus/google_cloud/manager.py]: where did this come from/where is it used? Seems like we shouldn't need that

**Pull request merged**